src/screenshot.cpp

```diff
@@ -856,3 +864,83 @@ bool MakeScreenshot(ScreenshotType t, const char *name)
 
 	return ret;
 }
+
+static byte _owner_colours[OWNER_END + 1];
+
+/**
+ * Return the colour a tile would be displayed with in the small map in mode "Owner".
+ *
+ * @param tile The tile of which we would like to get the colour.
+ * @return The colour of tile in the small map in mode "Owner"
+ */
+static inline byte GetMinimapPixels(TileIndex tile)
+{
+	Owner o;
+
+	switch (GetTileType(tile)) {
+	case MP_INDUSTRY: o = OWNER_END;          break;
+	case MP_HOUSE:    o = OWNER_TOWN;         break;
+	default:          o = GetTileOwner(tile); break;
+		/* FIXME: For MP_ROAD there are multiple owners.
+		 * GetTileOwner returns the rail owner (level crossing) resp. the owner of ROADTYPE_ROAD (normal road),
+		 * even if there are no ROADTYPE_ROAD bits on the tile.
+		 */
+	}
+
+	return _owner_colours[o];
+}
+
+static void MinimapScreenCallback(void *userdata, void *buf, uint y, uint pitch, uint n)
+{
+	uint32 *ubuf = (uint32 *)buf;
+
+	uint num = (pitch * n);
+	uint row, col;
+	byte val;
```

Comment:
again, declare variables with assignment - this isn't C, and even C's been able to assign variables for 20 years!